app/Models/Training/WaitingList.php

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * A Waiting List can be managed by many Staff Members (Accounts)
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function staff()
+    {
+        return $this->belongsToMany(Account::class, 'training_waiting_list_staff', 'list_id',
+            'account_id')->withTimestamps();
+    }
+
+    /**
+     * Many WaitingLists can have many Accounts (pivot).
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function accounts()
+    {
+        return $this->belongsToMany(Account::class, 'training_waiting_list_account',
+            'list_id')->using(WaitingListAccount::class)->withPivot(['position'])->withTimestamps();
+    }
+
+    /**
+     * Add an Account to a waiting list.
+     *
+     * @param Account $account | Collection
+     * @param Account $staffAccount
+     */
+    public function addToWaitingList($account, Account $staffAccount)
+    {
+        $this->accounts()->attach($account, ['position' => $this->nextPosition(), 'added_by' => $staffAccount->id]);
+    }
+
+    /**
+     * Remove an Account from a waiting list.
+     *
+     * @param Account $account
+     * @return void
+     */
+    public function removeFromWaitingList(Account $account)
+    {
+        $position = $this->accounts()->where('account_id', $account->id)->first()->pivot->position;
+
+        $this->accounts->transform(function ($item, $key) use ($position) {
+            if ($item->pivot->position > $position) {
+                return $item->pivot->decrementPosition();
+            }
+        });
+
+        $this->accounts()->detach($account);
+    }
+
+    /**
+     * Promote an Accounts' position within a WaitingList by the number passed.
+     *
+     * @param Account $account
+     * @param int $position
+     */
+    public function promote(Account $account, $position = 1)
+    {
+        if (!$this->accounts->contains($account)) {
+            throw new ModelNotFoundException($this);
+        } else {
+            $entry = $this->accounts()->where('account_id', $account->id)->first();
+
+            // current position before promotion
+            $oldPosition = $entry->pivot->position;
+
+            // deals with the actual promoted record; returns that new position
+            $newPosition = $entry->pivot->decrementPosition($position);
+
+            $this->accounts->transform(function ($item, $key) use ($oldPosition, $newPosition) {
+                if ($item->pivot->position < $oldPosition) {
+                    return $item->pivot->incrementPosition();
+                }
+            });
+        }
+    }
+
+    /**
+     * Demote an Accounts' position within a WaitingList by the number passed.
+     *
+     * @param Account $account
+     * @param int $position
+     */
+    public function demote(Account $account, $position = 1)
```

Comment:
Can we fire an event from this function to log it? Happy for you to just fire an event for now and then implement in another branch?

I'm not sure if we already have something defined already in our code base.

_(comment applicable for `promote` too)_